### DISCUSSION TITLE
Number helper

### DISCUSSION THREAD

**Discussion by allanmcarvalho** [laravel/framework#42941] [Ideas] [answered]
Hi guys!

I came from CakePHP and there I was since from CakePHP 2.x.

One of the things that I used everyday on cake was the Number helper. It's very useful to format currency, percentages, float numbers, prefixed and postfixed cases and other things.

I don't know if I just not found it on Laravel, or if it doesn't exist, but I think that this will be a big improvement on the framework core.

Thanks in advance!

**Comment by bonzai** [comment]
There is no such helper in Laravel, but number formatting is built-in into PHP:
https://www.php.net/manual/en/class.numberformatter.php

**Reply by allanmcarvalho** [reply]
I know and use number formatter.

But if we are to categorize everything by this principle, most of the things implemented in Laravel are natively available in PHP.  I'm talking here about doing with the numbers part what Laravel does very well, taking something that exists in PHP and making it even better.

If you've never used CakePHP's number helper, please take the time and try it out… I think you'll love it and understand better what I mean.

it's not just formatting numbers, it's making it easier, more “object oriented”, more intuitive, more integrated with the framework and its settings… I blindly believe that everyone wins!

**Reply by allanmcarvalho** [reply]
I see in Laravel a framework that has tools and its own impeccably built structure. Everything was thought of in the smallest details to be incredible.  It seems that it's not just a matter of “in PHP this already exists”.  Not! It has to be incredible… And that's what I see in the numbers part, which leaves something to be desired and that everyone would win if it existed!

**Reply by muhammadmp97** [reply]
What about using packages?

**Comment by allanmcarvalho** [comment]
https://github.com/cakephp/cakephp/blob/4.x/src/I18n/Number.php

It's a lot of possibilities that this helper can do in the context of "numbers".

**Comment by getsolaris** [comment]
```Cake\I18n\Number``` 

I found the document.
It would be useful if it was included in the helper.

**Comment by shaedrich** [comment]
Not strictly necessary but I do see your point compared to the bloated `Str` helper class

**Comment by smarcelloc** [comment, marked as answer]
In my opinion, the introduction of the `number` helper represents an advance for the Laravel community, especially having a part of the documentation that talks about formatting numbers, decimals and monetary values. This will simplify implementation, especially for less experienced developers, by formatting float values for your country's currency.

Furthermore, even experienced developers may not be entirely familiar with the `NumberFormatter`, and the need to instantiate an object for formatting can make the code more extensive. The inclusion of this helper eliminates the need for developers to create their own formatting classes or install third-party packages for this purpose.

The availability of the `NumberFormatter` class is excellent news for the development of this helper. By creating a helper to abstract the use of this native class, we make the lives of Laravel developers even easier when implementing this feature.

Personally, I'm excited about this idea and believe it will bring substantial benefits to the Laravel community.

**Reply by allanmcarvalho** [reply]
I agree 100% with your opinion! You mentioned many relevant points as to why this should be within Laravel. I a know what I say, because my highest demanding job is a crypto robot using Laravel, so, I work with numbers in whole application. It's a nightmare works with a native PHP Numbers functions to do simple things, like a currency format, and/or rounds, not to mention the floating point rounding solution that something cause some strange behaviors.

One more thing for example that can arrive with number(), is the round by step (I use every day). Eg, you can buy some crypto by lots under 0.005 steps and after calculate you know that can invest 12.996, the right lot will be 13.000 or 12.995 depending your round rule. Today I am like an expert in this areas of working with numbers due a many angry days LOL, but on begin was bad and very difficult...

@nunomaduro help us to score that great goal.
